apisix/discovery/nacos/init.lua

```diff
@@ -281,11 +283,18 @@ local function fetch_full_registry(premature)
                 up_apps[namespace_id]
                     [group_name][service_info.service_name] = nodes
             end
-            core.table.insert(nodes, {
+
+            local node = {
                 host = host.ip,
                 port = host.port,
                 weight = host.weight or default_weight,
-            })
+            }
+
+            if string.match(scheme, "grpc") and host.metadata.gRPC_port then
+                node.host = host.metadata.gRPC_port
+            end
```

Comment:
```suggestion
            local is_grpc = ngx_re_match(scheme, "grpc", "jo")
            if is_grpc and host.metadata and host.metadata.gRPC_port then
                node.host = host.metadata.gRPC_port
            end
```

and you need to add `local ngx_re_match       = ngx.re.match` at the header of file